solidity-verifiers/templates/nova_cyclefold_decider.askama.sol

```diff
@@ -144,4 +168,63 @@ contract NovaDecider is Groth16Verifier, KZG10Verifier {
 
         return(true);
     }
+
+    /**
+     * @notice  Verifies a Nova+CycleFold proof given initial and final IVC states, number of steps and the rest proof inputs concatenated.
+     * @dev     Simply reorganization of arguments and call to the `verifyNovaProof` function.
+     */
+    function verifyOpaqueNovaProofWithInputs(
```

Comment:
Could you add unit tests for `verifyOpaqueNovaProofWithInputs` and `verifyOpaqueNovaProof`, in a similar way to `nova_cyclefold::tests::nova_cyclefold_solidity_verifier`?

By the way, I ran these functions locally with `revm` and they worked great, and as a side question, do they also work on onchain EVMs? I am not sure if onchain EVMs have any stricter restrictions than `revm` on parameter length/stack size/etc.

Reply:
sure, sorry for skipping that initially - I wanted to firstly check if the change is okay with you, before writing tests :sweat_smile: 

now, we run old test scenarios with all 3 methods

Reply:
> do they also work on onchain EVMs

yes! I have deployed the contract to Aleph Zero EVM Testnet and called each of the function successfully - please check:

https://evm-explorer-testnet.alephzero.org/address/0xD96A7C3D1A45D6A01Ac2A50097FB1D95F9349Adf?tab=txs

Reply:
> > do they also work on onchain EVMs
> 
> yes! I have deployed the contract to Aleph Zero EVM Testnet and called each of the function successfully - please check:
> 
> https://evm-explorer-testnet.alephzero.org/address/0xD96A7C3D1A45D6A01Ac2A50097FB1D95F9349Adf?tab=txs

That's awesome! A huge thanks for experimenting with the new contract APIs. It is really nice to know everything works perfect onchain!